generative/maisi/scripts/diff_model_infer.py

```diff
@@ -0,0 +1,222 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import os
+import random
+import torch
+import torch.distributed as dist
+import nibabel as nib
+import numpy as np
+
+from datetime import datetime, timedelta
+from tqdm import tqdm
+
+from monai.inferers import sliding_window_inference
+from monai.utils import set_determinism
+
+from utils import define_instance, load_autoencoder_ckpt
+
+
+class ReconModel(torch.nn.Module):
+    def __init__(self, autoencoder, scale_factor):
+        super().__init__()
+        self.autoencoder = autoencoder
+        self.scale_factor = scale_factor
+
+    def forward(self, z):
+        recon_pt_nda = self.autoencoder.decode_stage_2_outputs(z / self.scale_factor)
+        return recon_pt_nda
+
+
+def diff_model_infer(env_config_path: str, model_config_path: str):
+    """
+    Main function to run the diffusion model.
+
+    Args:
+        env_config_path (str): Path to the environment configuration file.
+        model_config_path (str): Path to the model configuration file.
+    """
+    args = argparse.Namespace()
+
+    # Load environment configuration
+    with open(env_config_path, "r") as f:
+        env_config = json.load(f)
+
+    for k, v in env_config.items():
+        setattr(args, k, v)
+
+    # Load model configuration
+    with open(model_config_path, "r") as f:
+        model_config = json.load(f)
+
+    for k, v in model_config.items():
+        setattr(args, k, v)
+
+    a_min = -1000
+    a_max = 1000
+    b_min = 0
+    b_max = 1
+    print(f"a_min: {a_min}, a_max: {a_max}, b_min: {b_min}, b_max: {b_max}.")
+
+    dist.init_process_group(backend="nccl", init_method="env://", timeout=timedelta(seconds=7200))
+
+    local_rank = int(os.environ["LOCAL_RANK"])
+    world_size = int(dist.get_world_size())
+    device = torch.device("cuda", local_rank)
+    torch.cuda.set_device(device)
+    print(f"Using {device} of {world_size}")
+
+    # Set random seed for reproducibility
+    if args.diffusion_unet_inference["random_seed"] == None:
+        random_seed = random.randint(0, 99999)
+    else:
+        random_seed = args.diffusion_unet_inference["random_seed"] + local_rank
+    set_determinism(random_seed)
+    print(f"random seed: {random_seed}")
+
+    output_size = tuple(args.diffusion_unet_inference["dim"])
+    out_spacing = tuple(args.diffusion_unet_inference["spacing"])
+
+    ckpt_filepath = f"{args.model_dir}/{args.model_filename}"
+    output_prefix = args.output_prefix
+
+    if local_rank == 0:
```

Comment:
Maybe we use the monai RankFilter and the logging module to replace all the `print` in this function.

Reply:
Hi @mingxin-zheng, I have updated the script. Please proceed. Thanks.